### ISSUE TITLE
AG Grid lacks proper header and row role identification, failing 508

### ISSUE THREAD

**Issue by cmT293** [ceolter/ag-grid#9916]
### Describe the bug

**Current behavior**

Ag grid lacks identification of the row headers through the "role" property causing screen readers to not properly detect a collection of data as a data grid. This lack of identification fails the "1.3.1-table-identification" & "1.3.1-cell-header-association" 508 rules. The only way users can understand the relation of columns is through visual perception. 

**Expected behavior**

That the "row" and "columnheader" roles are used properly to define the grid columns/rows instead of the "presentation" role. This will help users identify the gird without visual perception. Also, roles should be something the developer can configure to meet the needs of their client or project.  

Here is documentation outlining the proper use of the roles and the 508 rules:

**Role examples** 
https://dequeuniversity.com/library/aria/table-responsive

**1.3.1-table-identification / 1.3.1-cell-header-association**
https://www.w3.org/TR/WCAG20-TECHS/F90.html

### Version

32.3.3

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers

**Comment by AG-Zoheil** [comment]
Hi,

We have not been able to reproduce your reported issue.

Looking at the grid in our demo, both role="row" and role="columnHeader" are present where they need to be.

![Image](https://github.com/user-attachments/assets/688e5e16-71d4-44d0-b90f-441ca0440a5b)

Kind regards,
Zoheil
